src/libsystemd/sd-event/sd-event.c

```diff
@@ -3664,7 +3664,7 @@ static int source_dispatch(sd_event_source *s) {
         if (s->n_ref == 0)
                 source_free(s);
         else if (r < 0)
-                sd_event_source_set_enabled(s, SD_EVENT_OFF);
+                (void) sd_event_source_set_enabled(s, SD_EVENT_OFF);
```

Comment:
turning an event source should always work. maybe enclose this in assert_se( >= 0);